File: ExpressionTreeTransform.Tests/CompilerGenerated/Method.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using Xunit;
using static ExpressionTreeTransform.Tests.Globals;
using static ExpressionTreeTransform.Tests.Runners;

namespace ExpressionTreeTransform.Tests {
    static class Dummy {
        internal static void DummyMethod() { }
    }

    [Trait("Source", CSharpCompiler)]
    public class Method {
        [Fact]
        public void InstanceMethod0Arguments() {
            var s = "";
            BuildAssert(
                () => s.ToString(),
                "() => s.ToString()",
                "Function() s.ToString"
            );
        }

        [Fact]
        public void StaticMethod0Arguments() {
            BuildAssert(
                () => Dummy.DummyMethod(),
                "() => Dummy.DummyMethod()",
                "Sub() Dummy.DummyMethod"
            );
        }

        [Fact]
        public void ExtensionMethod0Arguments() {
            var lst = new List<string>();
            BuildAssert(
                () => lst.Count(),
                "() => lst.Count()",
                "Function() lst.Count"
            );
        }

        [Fact]
        public void InstanceMethod1Argument() {
            var s = "";
            BuildAssert(
                () => s.CompareTo(""),
                "() => s.CompareTo(\"\")",
                "Function() s.CompareTo(\"\")"
            );
        }

        [Fact]
        public void StaticMethod1Argument() {
            BuildAssert(
                () => string.Intern(""),
                "() => string.Intern(\"\")",
                "Function() String.Intern(\"\")"
            );
        }

        [Fact]
        public void ExtensionMethod1Argument() {
            var lst = new List<string>();
            BuildAssert(
                () => lst.Take(1),
                "() => lst.Take(1)",
                "Function() lst.Take(1)"
            );
        }

        [Fact]
        public void InstanceMethod2Arguments() {
            var s = "";
            BuildAssert(
                () => s.Contains('a', StringComparison.InvariantCultureIgnoreCase),
                "() => s.Contains('a', StringComparison.InvariantCultureIgnoreCase)",
                "Function() s.Contains(\"a\"C, StringComparison.InvariantCultureIgnoreCase)"
            );
        }

        [Fact]
        public void StaticMethod2Arguments() {
            BuildAssert(
                () => string.Join(',', new[] { 'a', 'b' }),
                "() => string.Join(',', new [] {'a', 'b'})",
                "Function() String.Join(\",\"C, { \"a\"C, \"b\"C })"
            );
        }

        [Fact]
        public void ExtensionMethod2Arguments() {
            var lst = new List<string>();
            BuildAssert(
                () => lst.OrderBy(x=>x, StringComparer.OrdinalIgnoreCase),
                "() => lst.OrderBy((string x) => x, StringComparer.OrdinalIgnoreCase)",
                "Function() lst.OrderBy(Function(x As String) x, StringComparer.OrdinalIgnoreCase)"
            );
        }
    }
}

File: ExpressionToString.Tests.VB/CompilerGenerated/Method.vb
Friend Module Dummy
    Friend Sub DummyMethod()
    End Sub
End Module

Public Class Method
    <Fact>
    Sub InstanceMethod0Arguments()
        Dim s = ""
        BuildAssert(
            Function() s.ToString(),
            "() => s.ToString()",
            "Function() s.ToString"
        )
    End Sub

    <Fact>
    Sub StaticMethod0Arguments()
        BuildAssert(
            Sub() DummyMethod(),
            "() => Dummy.DummyMethod()",
            "Sub() Dummy.DummyMethod"
        )
    End Sub

    <Fact>
    Sub ExtensionMethod0Arguments()
        Dim lst As IEnumerable(Of String) = New List(Of String)()
        BuildAssert(
            Function() lst.Distinct,
            "() => lst.Distinct()",
            "Function() lst.Distinct"
        )
    End Sub

    <Fact>
    Sub InstanceMethod1Argument()
        Dim s = ""
        BuildAssert(
            Function() s.CompareTo(""),
            "() => s.CompareTo("""")",
            "Function() s.CompareTo("""")"
        )
    End Sub

    <Fact>
    Sub StaticMethod1Argument()
        BuildAssert(
            Function() String.Intern(""),
            "() => string.Intern("""")",
            "Function() String.Intern("""")"
        )
    End Sub

    <Fact>
    Sub ExtensionMethod1Argument()
        Dim lst As IEnumerable(Of String) = New List(Of String)()
        BuildAssert(
            Function() lst.Take(1),
            "() => lst.Take(1)",
            "Function() lst.Take(1)"
        )
    End Sub

    <Fact>
    Sub InstanceMethod2Arguments()
        Dim s = ""
        BuildAssert(
            Function() s.Contains("a"c, StringComparison.InvariantCultureIgnoreCase),
            "() => s.Contains('a', StringComparison.InvariantCultureIgnoreCase)",
            "Function() s.Contains(""a""C, StringComparison.InvariantCultureIgnoreCase)"
        )
    End Sub

    <Fact>
    Sub StaticMethod2Arguments()
        Dim arr As IEnumerable(Of Char) = New Char() {"a"c, "b"c}
        BuildAssert(
            Function() String.Join(","c, arr),
            "() => string.Join(',', arr)",
            "Function() String.Join("",""C, arr)"
        )
    End Sub

    <Fact>
    Sub ExtensionMethod2Arguments()
        Dim lst As IEnumerable(Of String) = New List(Of String)()
        Dim comparer As IComparer(Of String) = StringComparer.OrdinalIgnoreCase
        BuildAssert(
            Function() lst.OrderBy(Function(x) x, comparer),
            "() => lst.OrderBy((string x) => x, comparer)",
            "Function() lst.OrderBy(Function(x As String) x, comparer)"
        )
    End Sub

    <Fact>
    Sub StringConcat()
        BuildAssert(
            Function(s1 As String, s2 As String) String.Concat(s1, s2),
            "(string s1, string s2) => s1 + s2",
            "Function(s1 As String, s2 As String) s1 + s2"
        )
    End Sub

    <Fact>
    Sub StringConcatOperator()
        BuildAssert(
            Function(s1 As String, s2 As String) s1 + s2,
            "(string s1, string s2) => s1 + s2",
            "Function(s1 As String, s2 As String) s1 + s2"
        )
    End Sub

    <Fact>
    Sub StringConcatOperatorParamArray()
        BuildAssert(
            Function(s1 As String, s2 As String) s1 + s2 + s1 + s2 + s1 + s2,
            "(string s1, string s2) => s1 + s2 + s1 + s2 + s1 + s2",
            "Function(s1 As String, s2 As String) s1 + s2 + s1 + s2 + s1 + s2"
        )
    End Sub
End Class
